### PR TITLE
 Make react-router-dom treeshakable

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/ReactTraining/react-router/issues/6464

Here's the problem. Babel transpiles static class properties outside of class iife which blocks uglify/terser from this iife.

`defaultProps` will become a bad pattern soon since it's blocks treeshakability. For functions the problem is even more aggressive since it's not solvable these days.

I think the best we may do is use defaults in destructuring
```js
const Component = ({ prop1 = 'a', prop2 = 'b' }) => {
}
```

With classes this is harder because defaults can't be propagated in all methods.

### CODE REVIEW

File: packages/react-router-dom/.size-snapshot.json [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+    "gzipped": 1612,
     "treeshaked": {
       "rollup": {
-        "code": 1250,
+        "code": 453,
         "import_statements": 417
       },
       "webpack": {
-        "code": 3322
+        "code": 1661
```
Comment (timdorr) [inline packages/react-router-dom/.size-snapshot.json]: Nice!

File: packages/react-router-dom/rollup.config.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const babelOptionsESM = {
 const commonjsOptions = {
   include: /node_modules/,
   namedExports: {
-    "../react-router/node_modules/react-is/index.js": ["isValidElementType"]
+    "react-is": ["isValidElementType"]
```
Comment (timdorr) [inline packages/react-router-dom/rollup.config.js]: This change is breaking the build. Is it necessary? Do we need to bump react-is?

Comment (TrySound) [reply packages/react-router-dom/rollup.config.js]: Hm.. weird. This is broken for me locally.

Comment (TrySound) [reply packages/react-router-dom/rollup.config.js]: Reverted

**Comment by billyjanitsch** [comment]
Can you explain why this solution is preferred over simply not bundling the package into a single file for the ESM build? In other words, what is the advantage of bundling the file?

File: packages/react-router-dom/.size-snapshot.json (hunk 1 above)
Comment (mjackson) [reply packages/react-router-dom/.size-snapshot.json]: Cool 😎

**Comment by mjackson** [comment]
Thanks, @TrySound! I'll take a look at eliminating `defaultProps` from the `react-router` package.

**Pull request merged**